docs/build.adoc

```diff
@@ -177,29 +177,33 @@ ____
 `Step 1.` The following command will generate the VS project
 
 ----
-cmake -G"Visual Studio 15 2017 Win64" -S . -Bbuild/windows
+cmake -G "Visual Studio 15 2017 Win64" -S . -Bbuild/windows -DCMAKE_BUILD_TYPE=Release
```

Comment:
Why release?

Reply:
This setting, combined with some cmake code, sets the available build configs for Visual Studio and forces VS into building for Release vs. starting up in Debug mode.  If you leave this setting off, then cmake will leave all the build configs in place and VS will start up in Debug build mode - kind of annoying when you want to build for release.  Take a look at my proposed build.adoc changes and you will see what I mean.

Reply:
Starting in Debug mode isn't bad behavior.  I'd actually propose that release mode is almost never what you actually want when you're developing.  It's only useful when you want to build without symbols and ability to hit breakpoints or when you're ready to release something.  It's marginally faster for some use cases, but for a samples project where no one will release to the real world; I don't know if I see the usefulness of defaulting to release.
Then again, I'm in favor of defaulting to builds with as much instrumentation ready as I can find (LSAN etc).

Reply:
Ok thanks - good perspective.  However, the rest of the documentation is all about building for Release across all platform types.  So, I felt that Windows was an outlier and should be aligned.  However, if you mean that command line builds should always be Release, and VS and Xcode should always start up in Debug, then I understand and this could be reverted to support that strategy.

Reply:
I'm personally not a fan of suggesting the command line build types should be Release either.  However, if the developer is using commandline to compile, then it's trivial to switch that to Debug and get what they want.  I think in this case, we should default the IDEs to use Debug for development to encourage usage of tooling and the reasoning for having an IDE (i.e. demonstrate as much as possible how to debug/work in a Vulkan Project rather than something that's built for a product to release).

Reply:
Thanks for the feedback and reasoning.  This has now been reverted.